src/core/models/payloads/milestone.h

```diff
@@ -7,30 +7,62 @@
-                                                                 // IDs of all the not-ignored state-mutating
-                                                                 // transaction payloads referenced by the milestone.
   uint32_t
       next_pow_score;  // The new PoW score all messages should adhere to. If 0 then the PoW score should not change.
   uint32_t next_pow_score_milestone_index;  // The index of the first milestone that will require a new minimal pow
                                             // score for applying transactions. This field comes into effect only if the
-                                            // Next PoW score field is not 0.
-  UT_array *pub_keys;                       // Ed25519 Public Keys
-  void *receipt;                            // The inner payload of the milestone. Can be NULL or a Receipt.
+                                            // Next PoW Score field is not 0.
+} milestone_pow_option_t;
+
+/**
+ * @brief Milestone options types
+ *
+ */
+typedef enum {
+  MILESTONE_OPTION_RECEIPTS = 0,  // Defines dynamic changes to the PoW parameters.
+  MILESTONE_OPTION_POW,           // Defines UTXOs for newly migrated funds.
+} milestone_option_e;
+
+/**
+ * @brief Milestone option object
+ *
+ */
+typedef struct {
+  milestone_option_e type;  ///< The type of milestone option.
+  void *option;             ///< One of milestone options.
+} milestone_option_t;
+
+/**
+ * @brief A list of milestone options
+ *
+ */
+typedef struct milestone_options_list {
+  milestone_option_t *option;           // Point to current milestone option
+  struct milestone_options_list *next;  // Point to next milestone option
+} milestone_options_list_t;
+
+typedef struct {
+  uint32_t type;       // payload type, set to 7 denotes a milestone payload
+  uint32_t index;      // The index number of the milestone.
+  uint32_t timestamp;  // The Unix time (seconds since Unix epoch) at which the milestone was issued.
+  byte_t last_milestone_id[CRYPTO_BLAKE2B_256_HASH_BYTES];  // The milestone ID of the milestone with index number - 1
+  UT_array *parents;                                        // Parents of milestone message.
+  byte_t confirmed_merkle_root[CRYPTO_BLAKE2B_256_HASH_BYTES];  // The Merkle tree hash (BLAKE2b-256) of the message
+                                                                // IDs of all messages confirmed by this milestone.
+  byte_t applied_merkle_root[CRYPTO_BLAKE2B_256_HASH_BYTES];    // The Merkle tree hash (BLAKE2b-256) of the message IDs
+                                                              // of all messages applied by this milestone that contain
+                                                              // a state-mutating transaction.
+  byte_buf_t *metadata;               // Binary data only relevant to milestone issuer, e.g. internal state.
+  milestone_options_list_t *options;  // Milestone options: PoW milestone option or Receipts milestone option
```

Comment:
> atMostOneOfEach | At most one (none or one) of each of the listed subschemas.

it should be only one or none, therefore a list is not needed.

```suggestion
  milestone_options_t *option;  // Milestone options: PoW milestone option or Receipts milestone option
```

Reply:
`oneOf` means only one of listed items could be present.
`atMostOneOfEach` means all of listed items can be present but just one instance of each.

Example of `atMostOneOfEach` in `Blocks`:
https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#basic-output
Each of Blocks can be present but each of them just once.

Reply:
you are right. :)